main.js

```diff
@@ -10,42 +11,65 @@ import {
   getAssetSizes,
 } from './lib/helpers';
 
-let octokit;
-
 async function run() {
   try {
-    const myToken = getInput('repo-token', { required: true });
-    octokit = new GitHub(myToken);
+    const { myToken, cwd, usePrArtifacts } = getActionInputs();
+
+    const octokit = new GitHub(myToken);
+
     const pullRequest = await getPullRequest(context, octokit);
+    const fileDiffs = await diffAssets({ pullRequest, cwd, usePrArtifacts });
+
+    await commentOnPR({ octokit, pullRequest, fileDiffs })
+  } catch (error) {
+    setFailed(error.message);
+  }
+}
+
+async function getActionInputs() {
+  const workingDirectory = getInput('working-directory', { required: false });
+  const usePrArtifacts = getInput('use-pr-artifacts', { required: false });
+  const myToken = getInput('repo-token', { required: true });
 
-    const prAssets = await getAssetSizes();
+  const cwd = path.join(process.cwd(), workingDirectory);
 
-    await exec(`git checkout ${pullRequest.base.sha}`);
+  return { myToken, cwd, usePrArtifacts };
+}
+
+async function diffAssets({ pullRequest, cwd, usePrArtifacts }) {
+  const prAssets = await getAssetSizes({ cwd, build: !usePrArtifacts });
+
+  await exec(`git checkout ${pullRequest.base.sha}`, { cwd });
 
-    const masterAssets = await getAssetSizes();
+  const masterAssets = await getAssetSizes({ cwd, build: true });
```

Comment:
the master branch is always going to be rebuilt (no change with this PR). It'd be cool if there was a way to store artifacts per-branch somewhere (maybe that's already an option?) and then _only_ re-build if artifacts aren't found?